web/src/people.ts

```diff
@@ -1357,8 +1357,11 @@ export function get_mention_syntax(full_name: string, user_id?: number, silent =
         mention += "@**";
     }
     const wildcard_match = full_name_matches_wildcard_mention(full_name);
-    if (wildcard_match && user_id === undefined) {
-        mention += util.canonicalize_stream_synonyms(full_name);
```

Comment:
Because we still need this in place for generating URLs with "stream", simplified the logic here for the one case where we're changing the user's selected wildcard text.